docker/dev/Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM python:3.7-alpine3.10
+
+ENV PYTHONUNBUFFERED=0
+
+RUN apk add --no-cache curl
+
+RUN curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py | python \
+    && ln -s /root/.poetry/bin/poetry /usr/local/bin/poetry
+
+COPY ./docker/common /tmp/build
+
+RUN set -x \
+    && sh /tmp/build/install_build_deps.sh \
+    && sh /tmp/build/install_runtime_deps.sh
+
+COPY ./fonts/* /usr/share/fonts/
+
+RUN fc-cache -f -v
+
+COPY . /web
```

Comment:
why can't we install the requirements before copying the entire source code into the image? this way anytime we change a line and run the build it installs all the requirements anew. 

on another note: if we have a separate dockerfile for production and development, is it even useful to copy the source code into the image? won't we always run the development with source-code bind-mounted into the container?

Reply:
Poetry needs several files to be present for generating requirements.txt files.

I can try to only copy the necessary files instead.

Reply:
didn't work with poetry so idk, but I think you should only need the `pyproject.toml` and poetry's lock file

Reply:
yes and specify --no-root, so it does not require the actual package.